### PR TITLE
Remove the idea of extensions from Calamari

### PR DESCRIPTION
**Pull request by scottmerchant**
While porting kubernetes script steps to commands, we noticed that the extensions idea doesn't really make sense anymore as we always ship those assemblies anyway. We just need to ensure they are only loaded when calamari is not .net4.

### CODE REVIEW

File: source/Calamari.Aws/Integration/AwsScriptWrapper.cs [hunk 1]
```diff
@@ -16,7 +19,11 @@ public class AwsScriptWrapper : IScriptWrapper
         readonly ILog log;
         readonly IVariables variables;
         public int Priority => ScriptWrapperPriorities.CloudAuthenticationPriority;
-        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) => true;
+
+        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax) =>
+            variables.Get(SpecialVariables.Account.AccountType) == "AmazonWebServicesAccount" ||
+            string.Equals(variables.Get(AwsSpecialVariables.Authentication.UseInstanceRole), bool.TrueString,
+                StringComparison.InvariantCultureIgnoreCase);
```
Comment (scottmerchant) [inline source/Calamari.Aws/Integration/AwsScriptWrapper.cs:26]: This script wrapper would blow up if you didn't have the correct variables set with your AWS Creds (and previously it wouldn't be loaded if you didn't specify the AWS extension) so I had to put some checks in (which mirror those in Server in `WithAwsCalamariExtensionsIfRequired()` method.

File: source/Calamari.Tests/Fixtures/Manifest/ExecuteManifestCommandFixture.cs [hunk 2]
```diff
@@ -94,8 +92,7 @@ CalamariResult ExecuteCommand(VariableDictionary variables, string extensions =
                 return Invoke(Calamari()
                               .Action("execute-manifest")
                               .Argument("variables", variablesFile.FilePath)
-                              .Argument("sensitiveVariablesPassword", "GB8KdBqYRlgAON9ISUPdnQ==")
-                              .Argument("extensions", extensions));
```
Comment (scottmerchant) [inline source/Calamari.Tests/Fixtures/Manifest/ExecuteManifestCommandFixture.cs:98]: Turns out tests were using extensions to load the `Calamari.Tests` assembly so they could have test commands and stuff.. so I had to change some things around to get everything to work again.

File: source/Calamari.Aws/Integration/AwsScriptWrapper.cs (hunk 1 above)
Comment (MissedTheMark) [inline source/Calamari.Aws/Integration/AwsScriptWrapper.cs:26]: Felt a bit clunky to read this, might be worth modifying it to something like this:
```suggestion
        bool IScriptWrapper.IsEnabled(ScriptSyntax syntax)
        {
            var accountType = variables.Get(SpecialVariables.Account.AccountType);
            var useAwsInstanceRole = variables.Get(AwsSpecialVariables.Authentication.UseInstanceRole);
            return accountType == "AmazonWebServicesAccount" ||
                string.Equals(useAwsInstanceRole, bool.TrueString, StringComparison.InvariantCultureIgnoreCase);
         }
```

**Review by MissedTheMark — COMMENTED** [review]
Looks good to me, though I do feel that you should revert the program -> runner changes as we discussed.
I think I'd like to get @zentron to approve this though - lets see if we can get him to have a quick look at it this afternoon.

**Review by MissedTheMark — APPROVED** [review]
Approving, given that @zentron has approved the overall concept on the server side.

**Comment by zentron** [comment]
Looks good Thanks for breaking this up into its own piece.  🙇

**Pull request merged**